qiskit/primitives/base_sampler.py

```diff
@@ -0,0 +1,182 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+r"""
+=======
+Sampler
+=======
+
+Sampler class calculates probabilities or quasi-probabilities of bitstrings from quantum circuits.
+
+A sampler is initialized with the following elements.
+
+* quantum circuits (:math:`\psi_i(\theta)`): list of (parameterized) quantum circuits.
+  (a list of :class:`~qiskit.circuit.QuantumCircuit`))
+
+* parameters: a list of parameters of the quantum circuits.
+  (:class:`~qiskit.circuit.parametertable.ParameterView` or
+  a list of :class:`~qiskit.circuit.Parameter`).
+
+The estimator is run with the following inputs.
+
+* circuit indexes: a list of indices of the circuits to evaluate.
+
+* parameter values (:math:`\theta_k`): list of sets of parameter values
+  to be bound to the parameters of the quantum circuits.
+  (list of list of float)
+
+The output is a SamplerResult which contains probabilities or quasi-probabilities of bitstrings,
+plus optional metadata like error bars in the samples.
+
+The sampler object is expected to be closed after use or
+accessed within "with" context
+and the objects are called with parameter values and run options
+(e.g., ``shots`` or number of shots).
+
+Here is an example of how sampler is used.
+
+.. code-block:: python
+
+    from qiskit import QuantumCircuit
+    from qiskit.circuit.library import RealAmplitudes
+
+    bell = QuantumCircuit(2)
+    bell.h(0)
+    bell.cx(0, 1)
+    bell.measure_all()
+
+    # executes a Bell circuit
+    with Sampler(circuits=[bell], parameters=[[]]) as sampler:
+        result = sampler(parameters=[[]], circuits=[0])
+        print([q.binary_probabilities() for q in result.quasi_dists])
+
+    # executes three Bell circuits
+    with Sampler([bell]*3, [[]]) as sampler:
+        result = sampler([0, 1, 2], [[]]*3)
+        print([q.binary_probabilities() for q in result.quasi_dists])
+
+    # parametrized circuit
```

Comment:
To be consistent with other parts.

```suggestion
    # parameterized circuit
```